geopy/geocoders/geocodio.py

```diff
@@ -0,0 +1,243 @@
+            argument you must *not* provide a ``query`` argument.
+
+        :rtype: ``None``, :class:`geopy.location.Location` or a list of them, if
+            ``exactly_one=False``.
+        """
+        if query is not None and \
+                any(p is not None for p in (city, state, postal_code, country)):
+            raise GeocoderQueryError('Address component must not be provided if '
+                                     'query argument is used.')
+        if street is not None and \
+                not any(p is not None for p in (city, state, postal_code)):
+            raise GeocoderQueryError('If street is provided must also provide city, '
+                                     'state, and/or postal_code.')
+
+        api = '%s://%s%s' % (self.scheme, self.domain, self.geocode_path)
+
+        params = dict(
+            api_key=self.api_key,
+            q=query,
+            street=street,
+            city=city,
+            state=state,
+            postal_code=postal_code,
+            country=country,
+            limit=limit
+        )
+        params = {
+            k: v for k, v in params.items() if v is not None
+        }
+
+        url = "?".join((api, urlencode(params)))
+
+        logger.debug("%s.geocode: %s", self.__class__.__name__, url)
+        callback = partial(self._parse_json, exactly_one=exactly_one)
+        return self._call_geocoder(url, callback, timeout=timeout)
+
+    def reverse(
+        self,
+        query,
+        *,
+        exactly_one=True,
+        timeout=DEFAULT_SENTINEL,
+        limit=None
+    ):
+        """Return an address by location point.
+
+        :param str query: The coordinates for which you wish to obtain the
+            closest human-readable addresses
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+        :param int limit: The maximum number of matches to return. This will be reset
+            to 1 if ``exactly_one`` is ``True``.
+
+        :rtype: ``None``, :class:`geopy.location.Location` or a list of them, if
+            ``exactly_one=False``.
+        """
+        params = {
+            'q': self._coerce_point_to_string(query),
+            'api_key': self.api_key
+        }
+        if limit is not None:
+            params['limit'] = limit
+
+        api = '%s://%s%s' % (self.scheme, self.domain, self.reverse_path)
+
+        url = "?".join((api, urlencode(params)))
+
+        logger.debug("%s.reverse: %s", self.__class__.__name__, url)
+        callback = partial(self._parse_json, exactly_one=exactly_one)
+        return self._call_geocoder(url, callback, timeout=timeout)
+
+    @staticmethod
```

Comment:
Please make don't use staticmethod here. Those two are perfectly normal instance methods.

Reply:
Fixed in 5deb061124a933cc9b2d70c6db65e58a8bdf9b7b